crates/fluvio-smartengine/src/engine/wasmtime/engine.rs

```diff
@@ -157,8 +160,12 @@ impl SmartModuleChainInstance {
                     // encountered error, we stop processing and return partial output
                     return Ok(output);
                 } else {
-                    next_input = output.successes.try_into()?;
+                    next_input = SmartModuleInput::try_from_records(
+                        output.successes,
+                        DEFAULT_SMARTENGINE_VERSION,
```

Comment:
The decoding will happen using the version defined in `SmartModuleConfig` and passed to `SmartModuleInstanceContext`. We need somehow to use that version here, instead of hard-coded.

Reply:
yes, we should add test for it

Reply:
Here is an approach! https://github.com/infinyon/fluvio/pull/3389/commits/b9345e70a145df51a673211f692292892ebf6ee1